cmd/amass/main.go

```diff
@@ -237,8 +239,19 @@ func getLinesFromFile(path string) []string {
 		return lines
 	}
 	defer file.Close()
+	r = file
+	// Read the file as gzip if the file ends in gzip
+	if strings.HasSuffix(path, ".gz") {
```

Comment:
Maybe it would be more "robust" to check the actual MIME type of the opened file with [`DetectContentType`](https://golang.org/pkg/net/http/#DetectContentType) ?
Valid types would be `application/gzip` or `application/x-gzip`

Reply:
I could add it for a sanity check, no problem.

Reply:
I had a similar concern, and would prefer that the actual content be validated

Reply:
It would only read the first 512bytes of the file in order to determine its type.

Reply:
Sounds good, @c0bw3b